src/general.py

```diff
@@ -36,18 +37,114 @@ def get_resources() -> dict:
+            print("Error: Working dir must exist. Check arg val for -w and try again")
+            print("Exiting...")
+            sys.exit(1)
+
+    # -t option
+    # changes location of tests_dir, exiting if the dir dne
+    if args.change_test_dir == None:
+        tests_dir = Path("tests")
+    else:
+        tests_dir = Path(args.change_test_dir)
+        if not os.path.exists(tests_dir):
+            print("Error: Test dir must exist. Check arg val for -t and try again")
+            print("Exiting...")
+            sys.exit(1)
+
+    # -s option
+    # prints the list of currenlt available resources to the command line
+    if args.snapshot:
+        print("Here is a list of all currently available GLIDEIN_ResourceName s:")
+        resources = get_resources()
+        for resource in resources.keys():
+            print(resource)
+        print("End of resource list")
+
+    # -p option
+    # prints all the test names in tests_dir to the command line
+    if args.print_tests:
+        print("Here is a list of all tests in the test dir:")
+        for test in tests_dir.iterdir():
+            print(test.name)
+        print("End of test list")
+
+    # -f option
+    # clears the working_dir
+    if args.flush_all:
+        print("Flushing entire working directory")
+        for item in working_dir.iterdir():
+            shutil.rmtree(item)
+
+    # -d option
+    # clears all timestamp_dirs in working_dir older than the date provided
+    # will exit with an err if the date is not provided in the requested format
+    if (not args.flush_by_date == None) and (not args.flush_all):
+        print("Flushing working directory by date")
+        format_date = parse_date(args.flush_by_date)
+        for subdir in working_dir.iterdir():
+            subdir_date = datetime.strptime(subdir.name, "%Y-%m-%d_%H-%M")
+            if subdir_date < format_date:
+                shutil.rmtree(subdir)
+
+    # -b option
+    # controls whether the excersier runs. set to True by default
+    if args.run:
+        execute_tests(tests_dir, working_dir, args.tests)
+
+
+def parse_date(date_from_cla: str) -> str:
+    """
+    Usage: parse through date_time argument from the command line (option -d)
+    @param date_from_cla: date string as entered at the command line
+    @return: datetime formatted str representation of the date_from_cla
     """
-    if run:
-        execute_tests(tests_dir, working_dir)
+    DATETIME_FORMAT_OPTS = {"date": ["%Y", "%m", "%d"], "time": ["%H", "%M"]}
+
+    # YYYY-MM-DD_HH-MM
+    num_hyphens = date_from_cla.count("-")
+    if num_hyphens > 3 or (num_hyphens == 3 and "_" not in date_from_cla):
+        print(
+            f"Error: Invalid date time '{date_from_cla}' provided. Check arg val for -d"
+        )
+        print("Exiting...")
+        sys.exit(1)
+
+    # date format index
+    dfi = num_hyphens if num_hyphens == 3 else num_hyphens + 1
+
+    date_fmt = "-".join(DATETIME_FORMAT_OPTS["date"][:dfi])
```

Comment:
```suggestion
    date_fmt = "-".join(DATETIME_FORMAT_OPTS["date"][:min(num_hyphens+1, 3)])
```
I checked and there is a python `min()` function (unsurprisingly).